File: solutions/2352. Equal Row and Column Pairs/2352.py
# Time complexity: O(n^3)
# Space complexity: O(1)
class Solution:
  def equalPairs(self, grid: List[List[int]]) -> int:
    n = len(grid)
    ans = 0

    for i in range(n):
      for j in range(n):
        k = 0
        while k < n:
          if grid[i][k] != grid[k][j]:
            break
          k += 1
        if k == n:
          ans += 1

    return ans        

File: solutions/2352. Equal Row and Column Pairs/2352.rs
// Time complexity: O(n^3)
// Space complexity: O(1)
impl Solution {
    pub fn equal_pairs(grid: Vec<Vec<i32>>) -> i32 {
        let n = grid.len();
        let mut ans = 0;

        for i in 0..n {
            for j in 0..n {
                let mut k = 0;
                while k < n {
                    if grid[i][k] != grid[k][j] {
                        break;
                    }
                }
            }
        }
    }
}
